sklearn/utils/estimator_checks.py

```diff
@@ -553,6 +554,45 @@ def check_sample_weights_list(name, estimator_orig):
         estimator.fit(X, y, sample_weight=sample_weight)
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_sample_weight_invariance(name, estimator_orig):
+    if (has_fit_parameter(estimator_orig, "sample_weight") and
+                    name not in ["KMeans", "MiniBatchKMeans"]):
+        estimator1 = clone(estimator_orig)
+        estimator2 = clone(estimator_orig)
+
+        X = np.array([[1, 3], [1, 3], [1, 3], [1, 3],
+                      [2, 1], [2, 1], [2, 1], [2, 1],
+                      [3, 3], [3, 3], [3, 3], [3, 3],
+                      [4, 1], [4, 1], [4, 1], [4, 1]], dtype=np.dtype('float'))
+        y = np.array([1, 1, 1, 1, 2, 2, 2, 2,
+                      1, 1, 1, 1, 2, 2, 2, 2], dtype=np.dtype('float'))
+
+        if has_fit_parameter(estimator_orig, "random_state"):
```

Comment:
you can replace those using `set_random_state` from `sklearn.utils.testing` it will check if the estimators has already the random state

Reply:
Done!